src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -113,6 +148,24 @@ export const selectUnbondingEntriesByAccountSpecifier = createDeepEqualOutputSel
   }
 )
 
+export const selectAllUnbondingsEntriesByDenom = createDeepEqualOutputSelector(
+  selectStakingDataByAccountSpecifier,
+  selectDenom,
+  (stakingData, denom): Record<string, chainAdapters.cosmos.UndelegationEntry[]> => {
+    if (!stakingData || !stakingData.undelegations) return {}
+
+    return stakingData.undelegations.reduce((acc, { validator, entries }) => {
+      if (!acc[validator.address]) {
+        acc[validator.address] = []
+      }
+
+      acc[validator.address].push(...entries.filter(x => ASSET_ID_TO_DENOM[x.assetId] === denom))
```

Comment:
I used a spread because I need all elements to be added separately to the accumulator array, not just having a reference being added by the `push`. I could use a `concat` also but I think it would create a new array each time.